streams/src/test/java/org/apache/kafka/streams/KafkaStreamsTest.java

```diff
@@ -46,61 +46,99 @@ public void testStartAndClose() throws Exception {
         props.setProperty(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
         props.setProperty(StreamsConfig.METRIC_REPORTER_CLASSES_CONFIG, MockMetricsReporter.class.getName());
 
-        KStreamBuilder builder = new KStreamBuilder();
-        KafkaStreams streams = new KafkaStreams(builder, props);
+        final KStreamBuilder builder = new KStreamBuilder();
+        final KafkaStreams streams = new KafkaStreams(builder, props);
         streams.close();
         final int closeCount = MockMetricsReporter.CLOSE_COUNT.get();
 
         streams.close();
         Assert.assertEquals("subsequent close() calls should do nothing",
-                closeCount, MockMetricsReporter.CLOSE_COUNT.get());
+            closeCount, MockMetricsReporter.CLOSE_COUNT.get());
     }
 
-    @Test
+    @Test(expected = IllegalStateException.class)
     public void testCannotStartOnceClosed() throws Exception {
-        Properties props = new Properties();
+        final Properties props = new Properties();
         props.setProperty(StreamsConfig.APPLICATION_ID_CONFIG, "testCannotStartOnceClosed");
         props.setProperty(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
 
-        KStreamBuilder builder = new KStreamBuilder();
-        KafkaStreams streams = new KafkaStreams(builder, props);
+        final KStreamBuilder builder = new KStreamBuilder();
+        final KafkaStreams streams = new KafkaStreams(builder, props);
         streams.close();
 
         try {
             streams.start();
-        } catch (IllegalStateException e) {
+        } catch (final IllegalStateException e) {
             Assert.assertEquals("Cannot restart after closing.", e.getMessage());
-            return;
+            throw e;
+        } finally {
+            streams.close();
         }
-        Assert.fail("should have caught an exception and returned");
     }
 
-    @Test
+    @Test(expected = IllegalStateException.class)
     public void testCannotStartTwice() throws Exception {
-        Properties props = new Properties();
+        final Properties props = new Properties();
         props.setProperty(StreamsConfig.APPLICATION_ID_CONFIG, "testCannotStartTwice");
         props.setProperty(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
 
-        KStreamBuilder builder = new KStreamBuilder();
-        KafkaStreams streams = new KafkaStreams(builder, props);
+        final KStreamBuilder builder = new KStreamBuilder();
+        final KafkaStreams streams = new KafkaStreams(builder, props);
         streams.start();
 
         try {
             streams.start();
-        } catch (IllegalStateException e) {
+        } catch (final IllegalStateException e) {
             Assert.assertEquals("This process was already started.", e.getMessage());
-            return;
+            throw e;
+        } finally {
+            streams.close();
+        }
+    }
+
+    @Test
+    public void testCleanup() throws Exception {
```

Comment:
A positive test (ie, that it does work) does not require a complex name, IMHO. Describing test condition via naming makes only sense for negative tests (or if there are multiple different positive test -- there is just one in our case).